cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -179,13 +180,16 @@ private Injector init() throws Exception {
           @Override
           protected void configure() {
             bind(DatasetFramework.class).to(InMemoryDatasetFramework.class).in(Scopes.SINGLETON);
+            // the DataSetsModules().getDistributedModules() binds to RemoteDatasetFramework so override that to
+            // InMemoryDatasetFramework
+            bind(DatasetFramework.class)
```

Comment:
No I don't think so because this dsFramework is named and also in upgrade we always wants InMem. See the other binding earlier.